### ISSUE TITLE
Unable to clone repository with submodule url git@github.com:

### ISSUE THREAD

**Issue by karthikkolli** [go-git/go-git#397]
When using git:// as a submodule path, the clone fails. Current implementation assumes that submodule paths can either be a http url or relative url.
Source: https://github.com/go-git/go-git/blob/master/submodule.go#L136

Tested sample
error: parse "git@github.com:karthikkolli/subtest_module.git": first path segment in URL cannot contain colon
```
r, err := git.PlainClone("/tmp/subtest", false, &git.CloneOptions{
  URL:               "https://github.com/karthikkolli/subtest",
  RecurseSubmodules: git.DefaultSubmoduleRecursionDepth,
  Auth:              auth,
})
```

**Comment by rohankmr414** [comment]
@mcuadros This issue has been fixed on our fork of go-git in https://github.com/argoproj-labs/go-git and we're planning to switch back to upstream go-git, is there any update on this issue?

**Comment by rohankmr414** [comment]
cc: @pjbgf

**Comment by pjbgf** [comment]
@rohankmr414 I am waiting for https://github.com/go-git/go-git/pull/756 to be squashed, which should fix this issue. Would you be able to confirm whether those changes resolve the issues you are experiencing as well?

**Comment by rohankmr414** [comment]
@pjbgf #464 should fix it, we have it merged on our fork.

cc: @Hunter-Thompson

**Comment by pjbgf** [comment]
This should now be fixed as per the merge of #756.
